res/app/device-list/details/device-list-details-directive.js

```diff
@@ -277,6 +279,20 @@ module.exports = function DeviceListDetailsDirective(
         }
       })()
 
+      // check if new column needs a new scope or not
+      // and build accordingly
+      function buildColumn(col, device) {
```

Comment:
you were right, there are performance issues now. I tried with 1000 fake devices and it was taking pretty longer time to load compare to master branch.

I will try to fix these issues.